core/trino-main/src/main/java/io/trino/server/QueriesResource.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.server;
+
+import io.airlift.log.Logger;
+import io.trino.dispatcher.DispatchManager;
+import io.trino.server.security.ResourceSecurity;
+
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.server.security.ResourceSecurity.AccessType.PUBLIC;
+import static java.util.Objects.requireNonNull;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+// QueriesResource expose coordinator endpoint to retrieve pending and recently ended queries.
+//   /v1/queries/{maxEndAgeSec} --- list of BasicQueryInfo of pending and recently ended queries;
+@Path("/v1/queries")
+public class QueriesResource
+{
+    private static Logger log = Logger.get(QueriesResource.class);
+
+    private final DispatchManager dispatchManager;
+
+    @Inject
+    public QueriesResource(DispatchManager dispatchManager)
+    {
+        log.info("Construct QueriesResource");
+        this.dispatchManager = requireNonNull(dispatchManager, "dispatchManager is null");
+    }
+
+    // Get BasicQueryInfo of all pending and recently ended queries.
+    // Here recently is defined as ended on or after maxEndAgeSec seconds ago.
+    @ResourceSecurity(PUBLIC)
```

Comment:
IDK if it is ok to expose query list as `PUBLIC`. @dain/@electrum/@findepi can you comment here please?

Reply:
correct, query list cannot be public

Reply:
Changed to AUTHENTICATED_USER (despite there are more other AccessType).

Reply:
done